elastic/kibana
# Remove broadcast-channel dependency from security plugin

## Summary

Removes the `broadcast-channel` dependency from the security plugin. This was used to synchronize session state across multiple tabs.

This PR switches to the native [`BroadcastChannel`](https://developer.mozilla.org/en-US/docs/Web/API/BroadcastChannel), which is available in all of our supported browsers.

Resolves https://github.com/elastic/kibana/issues/127921

Tested locally in Edge `101.0.1210.47` and Safari `15.5`

## Testing

1) Configure a session timeout via `xpack.security.session.idleTimeout: 6m`
2) Login to Kibana using the basic auth provider
3) Open another tab, and display them side-by-side
4) Wait ~60 seconds
5) When the toast warning appears about an upcoming session expiration, click the button to extend your session in one of your tabs
6) Observe that the toast warning disappears automatically in the other tab, and your session remains active in both.
